onnxruntime/core/providers/cpu/tensor/transpose.cc

```diff
@@ -52,35 +94,39 @@ static void DoTransposeImpl(int64_t num_axes, const std::vector<int64_t>& target
                             size_t num_blocks, size_t num_elts_in_block, const std::vector<size_t>& stride,
                             const uint8_t* source, uint8_t* target, size_t element_size) {
   size_t blocksize = num_elts_in_block * element_size;
+  ORT_ENFORCE(num_axes > 0, "Transpose not implemented for empty tensors.");
+  MultiIndex* mindex = (MultiIndex*)alloca(num_axes * sizeof(MultiIndex));
```

Comment:
Is this ORT_ENFORCE necessary? Tranpose::Compute handles empty tensors with

```
  if (output_shape.Size() == 0)
    return Status::OK();
```

ORT_ENFORCE involves an 'if' and a throw so we should avoid using it unless required.